### DISCUSSION TITLE
Problem in my inlay_hints config for the new 0.10.0

### DISCUSSION THREAD

**Discussion by AllanJRY** [neovim/neovim#28806] [Q&A] [answered]
Hello ! 

First congratulations to everyone who contributed to the 0.10.0 release !

I'm reaching you about a configuration error which I face with inlay_hints. 

I based my configuration on kickstart.nvim and added those 2 blocks of code : 

In the init.lua : 
```lua
if vim.lsp.inlay_hint then
  vim.keymap.set(
  "n",
    "<leader>ih",
    function() vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled({}), {}) end,
    {desc = "Toggle Inlay Hints"}
  )
end
```

and in my after/ftplugin/rust.lua : 
```lua
-- enable inlay hints by default in rust files.
local bufnr = vim.api.nvim_get_current_buf()
if vim.lsp.inlay_hint then
    vim.api.nvim_create_autocmd('LspAttach', {
        callback = function(args)
            -- Wait for LSP server to start (just to be sure)
            vim.defer_fn(function() vim.lsp.inlay_hint.enable(true, {bufnr = bufnr}) end, 1000)
        end,
    })
end
```
(if statements are here because my config is also used in previous neovim versions)

When I open a Rust file I get this error : 
```
Error executing vim.schedule lua callback: ...les/Neovim/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:377: filter: expected table, got number                                                                
stack traceback:                                                                                               
    [C]: in function 'error'  
    vim/shared.lua:936: in function 'validate'                                                             
    ...les/Neovim/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:377: in function 'is_enabled'  
    ...ata/lazy/rustaceanvim/lua/rustaceanvim/server_status.lua:21: in function 'handler'
    ...m Files/Neovim/share/nvim/runtime/lua/vim/lsp/client.lua:1005: in function 'fn'                     
    vim/_editor.lua:351: in function <vim/_editor.lua:350>                                         
```

What I tried : 
- Passing nil values were Filter is required
- Comment one block, then the other and finally both, and the error still show.
- I verify every place in my configuration but those two blocks are the only two place where inlay_hints api is called, and it seems that I'm not passing number values.

I'm a beginner with neovim configuration so did I missed something ? 

Thank you for your help !

> **Note**: that doesn't prevent inlay_hints from working, they are correctly displayed, and I can toggle them with `<leader>ih` like I configured.

**Comment by zeertzjq** [comment, marked as answer]
Your rustaceanvim plugin version may be outdated.

**Reply by AllanJRY** [reply]
Oh yes I think you are right, I find a similar discussion on their repo ! 

Thank you and sorry ! I Should have check this before asking
